v2/tests/log_context_property_boolean_type_int/log_context_property_boolean_type_int.c

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/log_context_property_type.h"
+#include "c_logging/log_context_property_type_if.h"
+#include "c_logging/log_context_property_boolean_type.h"
+
+#define TEST_BUFFER_SIZE 1024
+
+#define POOR_MANS_ASSERT(cond) \
+    if (!(cond)) \
+    { \
+        (void)printf("%s:%d test failed\r\n", __FUNCTION__, __LINE__); \
+        abort(); \
+    } \
+
+/* LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).to_string */
+
+static void bool_to_string_with_NULL_value_fails(void)
+{
+    // arrange
+    char buffer[TEST_BUFFER_SIZE];
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).to_string(NULL, buffer, TEST_BUFFER_SIZE);
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+}
+
+static void bool_to_string_succeeds(void)
+{
+    // arrange
+    char buffer[TEST_BUFFER_SIZE];
+    bool bool_value = 0;
```

Comment:
>0 [](http://example.com/codeflow?start=22&length=1)

copy/pasta detected from the int types. I think we want to test with true/false primarily

Reply:
modified